apis/apps/v1alpha1/cluster_types.go

```diff
@@ -382,18 +396,27 @@ type ClusterComponentSpec struct {
 	// +kubebuilder:validation:MaxLength=22
 	// +kubebuilder:validation:Pattern:=`^[a-z]([a-z0-9\-]*[a-z0-9])?$`
 	// TODO +kubebuilder:validation:XValidation:rule="self == oldSelf",message="componentDefRef is immutable"
+	// +kubebuilder:deprecatedversion:warning="This field has been deprecated since 0.9.0, consider using the ComponentDef instead"
 	// +optional
 	ComponentDefRef string `json:"componentDefRef,omitempty"`
 
 	// References the name of the ComponentDefinition.
 	// If both componentDefRef and componentDef are provided, the componentDef will take precedence over componentDefRef.
 	//
-	// +kubebuilder:validation:MaxLength=22
+	// +kubebuilder:validation:MaxLength=128
```

Comment:
The ComponentDef is the name of ComponentDefinition. In K8s, the maximum length for the name of an object should be 64 characters.

Reply:
fixed